deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/DeploymentScannerService.java

```diff
@@ -178,6 +178,7 @@ public DeploymentOperations create() {
      */
     @Override
     public synchronized void stop(StopContext context) {
+        context.asynchronous();
```

Comment:
I am not super familiar with this but the usage pattern elsewhere seems to be
1) inject an executor
2) Call context.asynchronous()
3) Perform the work which may block in a Runnable
4) Call context context.complete() at the end of the Runnable

i.e. the heavy work should happen in a different thread.